docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,778 @@
+     @end
+
+     Note that you do *not* provide the `.norg` extension within the path.
+     You can use traditional modifiers in your path, like `/` (in e.g. `/my/file`) to signify
+     the root of your filesystem, `~` (in e.g. `~/Documents/my-file`) to signify the current
+     user's home directory, /or/ you can use the [Neorg]-specific `$` (in e.g. `$/my/file`) to
+     signify the _root_ of the [Neorg] workspace.
+     Since not all _Norg_ files will be used strictly by [Neorg], the workspace root can be
+     implementation-specific - for git repos the workspace root could be simply the root of the
+     repository, and for other note taking apps it could simply be the root of the directory where
+     all the notes are stored.
+
+     A file location may /only/ be accompanied by a {# detached modifier} or the
+     {# The Magic Char (`#`)}[the magic char], in which case the link looks like so:
+
+     @code norg
+     {:path/to/file:# Location within that file}
+     {:path/to/file:** Level 2 heading}
+     @end
+
+     `@` and URLs are not allowed:
+
+     @code norg
+     {:path:@ file} <- invalid
+     {:path:https://my-url} <- also invalid
+     @end
+
+**** URL
+     You can define a link to an external resource by simply putting in the URL:
+     
+     @code norg
+     {https://github.com/nvim-neorg/neorg}
+     @end
+
+**** Detached Modifier
+     Neorg allows you to link to any {# Structural Detached Modifiers}[structural] or
+     {# Range-able Detached Modifiers}[range-able] detached modifier:
+
+     @code norg
+     * I am a level 1 heading
+
+     Strict link to a level 1 heading:
+     {* I am a level 1 heading}
+     @end
+
+     The inside of the link location looks just like a detached modifier definition, and that's
+     because it pretty much is. You can substitute the `*` char for any other
+     {# Structural Detached Modifiers}[structural] or {# Range-able Detached Modifiers}[range-able]
+     detached modifier, /except/ "ranged" versions of said modifiers - those are *disallowed* within
+     the link syntax. By this we mean that syntax like `{$$ Text}` is _invalid_. To link to a ranged
+     definition you'd still use `{$ Text}`, there's no reason to make a distinction between a ranged
+     and non-ranged detached modifier, as both have the same meaning, one just allows more content
+     to exist within itself.
+
+**** Custom Detached Modifiers
+     Apart from linking to the detached modifiers outlined above, you can also link to a set of
+     custom modifiers specifically designed for links. These are the `#` (magic) and `@` (file)
+     linkable locations.
+
+***** The File Linkable (`@`)
+      Sometimes you may want to link to an external file that isn't an _Norg_ file.
+      In that case you may use `{@ /path/to/my/file.txt}` (notice the mandatory space after the `@`
+      char, just like with the {**** detached modifier}s). Paths are relative to the _Norg_
+      file that contains the link.
```

Comment:
No I'm pretty sure we wanted this syntax from the very beginning lol

Even the treesitter tests had `{@ this.txt}` as a valid and passing case.
It wouldn't make sense to use `{@myfile.txt}`, since that would make it the only char that is unnecessarily inconsistent with everything else (even the magic char, `#`, requires whitespace).